### DISCUSSION TITLE
Effects of changing heading tag for related post

### DISCUSSION THREAD

**Discussion by kulbhushanchand** [mmistakes/minimal-mistakes#3040] [Q&A] [answered]
Presently the [related posts section](https://github.com/mmistakes/minimal-mistakes/blob/cc9d1fdaf3a085e029c4d2e126816172505cce63/_layouts/single.html#L68) have `h4` tag. When the post contents does not have a `h3` heading the lighthouse gives SEO suggestion as below 
![image](https://user-images.githubusercontent.com/17742733/124411148-9a011980-dd69-11eb-9274-bf6361fdfe87.png)

I can change it to `h1` and `h2` as a workaround. 
**Is it intentional to kept the section heading at `h4`?**
**Is my workaround good enough and semantically correct?**

**Comment by kulbhushanchand** [comment, marked as answer]
After some research, I found there should be only one `h1` as a good practice. 
So, in the theme, the `h4` of the related posts section can be replaced by `h2` for the lighthouse SEO suggestion. 
Speaking of why original theme uses `h4` for related posts section, I don't know.

**Reply by iBug** [reply]
I guess H4 was used mainly for cosmetic reasons, i.e. its size looks good for that place.

I believe the difference between H2 and H4 carries no semantic significance here, as "related posts" isn't really part of the article content, just like the table of content. I'd rather say this specific SEO suggestion is one of those you'd better ignore than take seriously.

**Reply by ryan-p-randall** [reply]
I think `H2` would be better here, particularly for accessibility purposes. https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements#accessibility_concerns

The "related articles" section is a "sibling" to the article, not a "child" of any content in the article. `H4` would incorrectly present the "related articles" section as a "child" of the last `H2` or `H3` in the article's content.

So I'd strongly suggest using `H2` and then using CSS to make the heading display smaller.

**Reply by iBug** [reply]
I think it's much less of an issue as the article is already wrapped inside an `<article>` tag. The related posts are in a separate `<div>` so the heading level doesn't matter as much. Think more of it like some decoration than the content.

**Reply by ryan-p-randall** [reply]
Are the `<article>` and `<div>` tags announced by all screen reader programs? I don't see `aria-labelledby` anywhere, which I think is necessary to expose these to screen readers.

The simplest solution is usually to have the heading elements carry a logical order.

**Reply by mmistakes** [reply]
@iBug is right, the `h4` is likely more of a decoration thing that came from an older version of the theme before some work was done to make the HTML more semantic and accessible.

There definitely are accessibility and SEO concerns when headings aren't marked up in order, but I'm not sure how big of an issue it is when said heading falls outside of the main content... it's technically outside of `<article>` and treated more as an aside.

I don't have a strong opinion leaving it as is or replacing it. If it can be switched to a `<h2>` as @ryan-p-randall suggests and styled so it matches how it looks now... I'm cool with that.

**Reply by kulbhushanchand** [reply]
@mmistakes @ryan-p-randall 
There is no need for additional styling since the class for related post already provides the smaller font size.

**Reply by mmistakes** [reply]
Happy to review a PR if someone wants to tackle this change.

**Reply by kulbhushanchand** [reply]
Hi @mmistakes,
I have submitted the [relevant PR](https://github.com/mmistakes/minimal-mistakes/pull/3064).
